test/any/FetchSpecRunner-test.js

```diff
@@ -0,0 +1,186 @@
+
+            map[this.MICROSYNTAXES_URL] = this.HTTPResponse.create({
+              status: 200,
+              payload: Promise.resolve(this.MICROSYNTAXES_CONTENT)
+            });
+
+            map[this.FETCHING_URL] = this.HTTPResponse.create({
+              status: 200,
+              payload: Promise.resolve(this.FETCHING_CONTENT)
+            });
+
+            return map;
+          }
+        }
+      ],
+
+      methods: [
+        function send() {
+          var response = this.urlMap[this.url];
+          if (response !== undefined) {
+            return Promise.resolve(response);
+          } else {
+            // Mock a failure HTTP Request.
+            return Promise.resolve(this.HTTPResponse.create({
+              status: 404,
+              payload: Promise.resolve('The request resource was not found')
+            }));
+          }
+        }
+      ]
+    });
+
+    foam.CLASS({
+      package: 'org.chromium.webidl.Test',
+      name: 'ResultBox',
+      implements: ['foam.box.Box'],
+
+      properties: [
+        {
+          class: 'Array',
+          name: 'results',
+        },
+      ],
+
+      methods: [
+        function send(msg) {
+          this.results.push(msg);
+        },
+        function clear() {
+          this.results = [];
+        }
+      ]
+    });
+
+    FetchSpecRunner = foam.lookup('org.chromium.webidl.FetchSpecRunner');
+    PipelineMessage = foam.lookup('org.chromium.webidl.PipelineMessage');
+    ResultBox = foam.lookup('org.chromium.webidl.Test.ResultBox');
+    var MockHTTPRequest = foam.lookup('org.chromium.webidl.test.MockHTTPRequest');
+    foam.register(MockHTTPRequest, 'foam.net.RetryHTTPRequest');
+  });
+
+  it('should send an error if invalid arguments are received as a message', function() {
+    var outputBox = ResultBox.create();
+    var errorBox = ResultBox.create();
+    var runner = FetchSpecRunner.create({
+      outputBox: outputBox,
+      errorBox: errorBox,
+    });
+
+    // Sending any object other than a PipelineMessage object should
+    // result in a message sent to errorBox.
+    var wrongObj = {};
+    runner.run(wrongObj);
+    expect(errorBox.results.length).toBe(1);
+
+    // Sending a PipelineMessage that is missing urls field should
+    // result in a message sent to errorBox.
+    var msg = PipelineMessage.create();
+    runner.run(msg);
+    expect(errorBox.results.length).toBe(2);
+  });
+
+  it('should spawn a worker and fetch the given URLs', function(done) {
+    var outputBox = ResultBox.create();
+    var errorBox = ResultBox.create();
+    var runner = FetchSpecRunner.create({
+      outputBox: outputBox,
+      errorBox: errorBox,
+    });
+
+    var urls = [
+      'https://html.spec.whatwg.org/multipage/common-microsyntaxes.html',
+      'https://html.spec.whatwg.org/multipage/urls-and-fetching.html',
+    ];
+    var msg = PipelineMessage.create({ urls: urls });
+    runner.run(msg);
+
+    // Check back in a second to allow for fetching.
```

Comment:
Perhaps we should add a `TODO`; monitoring output and error boxes on the component can be used to know when it is safe to trigger expectations.